Add ability to add additional labels to server pods

peterklijn opened the pull request:
I would like to add the ability to add additional labels to consul-server pods, the same way annotations can be added.

At the company I work for, we use specific Kubernetes pod labels to tell Fluentd which Elasticsearch index should be used to send logs to. With this change I can the additional label :)

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/consul-helm?pullRequest=553)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/consul-helm?pullRequest=553)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/consul-helm?pullRequest=553).</sub>

ashwin-venkatesh commented on test/unit/server-statefulset.bats:
```diff
@@ -345,6 +345,28 @@ load _helpers
+  [ "${actual}" = "{}" ]
+}
+
+@test "server/StatefulSet: pod labels can be set" {
```
Could you add another testcase here which would verify multiple labels getting set if `podLabels` has multiple values set?

peterklijn replied:
Sure thing :)

ashwin-venkatesh commented on values.yaml:
```diff
@@ -321,6 +321,13 @@ server:
+  # Example:
+  #   podLabels: |
+  #     "label-key": "label-value"
+  podLabels: null
```
Can this be renamed to `extraLabels`? That seems to be aligned with how we have addressed similar constructs in the values file. see: extraConfig, extraVolumes.

Additionally, can you update the example to demonstrate setting multiple labels? It should help other users when they are trying to set this up

peterklijn replied:
Done :)

peterklijn commented:
No problem @ashwin-venkatesh :)

I've updated the `extraLabels` to use a yaml block.

Additionally I've updated the `annotations` to also support a yaml block (it currently supports both a yaml block and a string which is documented as deprecated). Let me know if that's ok :)

lkysow commented:
I think we should leave out the annotations as string deprecation from this PR and put it in a separate PR that does the deprecation across all places where annotations can be set so we don't have inconsistency across the chart between components.

peterklijn commented:
Thanks for the review @ashwin-venkatesh and @lkysow,

I've removed the annotation change and rebased with upstream
